valgrind.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+# cd to the script directory:
+cd "${0%/*}" || { echo "Couldn't cd to ${0%/*}!"; exit 1; }
+VALGRIND="valgrind"
+if ! [ -x "$(command -v ${VALGRIND})" ]; then
+  echo "{$VALGRIND} not found. skipping..."
+  exit 1
+fi
+
+valgrind --error-exitcode=1 --tool=memcheck --leak-check=full --track-origins=yes -q --error-limit=no --gen-suppressions=all $1
```

Comment:
Doesn't shellcheck complain about this?
```suggestion
valgrind --error-exitcode=1 --tool=memcheck --leak-check=full --track-origins=yes -q --error-limit=no --gen-suppressions=all "$@"
```

That said, I have my doubts about the usefulness of this wrapper script. When run from CI, they can simply be part of gitlab-ci.yml. When run manually, there are a lot of them that I may not want at all.

Reply:
Yes, shellcheck was complaining, my bad. Fixed!

When this script is run manually, you must provide the name of the test application as a parameter. And you can also provide a filter expression argument to specify which test from within such application should run. Example:
```
./valgrind.sh ./build/out/bin/bwl_dummy_unit_tests --gtest_filter=nl80211_client_test.get_channel_utilization_should_succeed_*
```

See [Running Test Programs: Advanced Options](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#running-test-programs-advanced-options) for a description of all available command line flags.

You can also provide extra arguments to Valgrind before the name of the test application.

Maybe we could have created an Alias Command instead.